skops/io/_sklearn.py

```diff
@@ -146,6 +173,13 @@ def Tree_get_instance(state, src):
     return reduce_get_instance(state, src, Tree)
 
 
+def sgd_loss_get_instance(state, src):
+    cls = gettype(state)
+    if cls not in ALLOWED_SGD_LOSSES:
```

Comment:
Why do we need to check this, wouldn't the dispatcher make sure we only get valid classes? And if we need to check it, could we not check if the class is a subclass of `LossFunction` instead?

Reply:
I'm trying to prevent this here:

```python
class DarkLord(LossFunction):
	def __init__(self, command):
		exec(command)
	def __reduce__(self):
		return (DarkLord, self.command)
```

we still don't accept custom code/classes, but we need to figure that out anyway, and thought this would kinda make it slightly safer.

I'll send a PR to remove these from sklearn anyway.

Reply:
> we still don't accept custom code/classes, but we need to figure that out anyway, and thought this would kinda make it slightly safer.

Yes, IIUC, the sketched attack can only work if the user has already installed a package containing the `DarkLord` class, right? In which case they're already compromised.

Reply:
yes, right now that's correct. That's why:

> we still don't accept custom code/classes, but we need to figure that out anyway

So it wouldn't be just a package, it can be code inside the persisted model.